components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/sshuseraccesschecktrigger/sshuseraccesschecktrigger.go

```diff
@@ -0,0 +1,57 @@
+package sshuseraccesschecktrigger
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/constants"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/utils/configutils"
+	"github.com/chef/automate/lib/logger"
+)
+
+type SshUserAccessCheck struct {
+	host string
+	port string
+	log  logger.Logger
+}
+
+func NewSshUserAccessCheck(log logger.Logger, port string) *SshUserAccessCheck {
+	return &SshUserAccessCheck{
+		log:  log,
+		host: constants.LOCAL_HOST_URL,
+		port: port,
+	}
+}
+
+func (ss *SshUserAccessCheck) Run(config models.Config) []models.CheckTriggerResponse {
+
+	ss.log.Info("Performing SSH user access check from batch check ")
+	count := config.Hardware.AutomateNodeCount + config.Hardware.ChefInfraServerNodeCount +
+		config.Hardware.PostgresqlNodeCount + config.Hardware.OpenSearchNodeCount
+
+	outputCh := make(chan models.CheckTriggerResponse, count)
+	url := fmt.Sprintf("%s:%s%s", ss.host, ss.port, constants.SSH_USER_CHECK_API_PATH)
+
+	var finalResult []models.CheckTriggerResponse
+	hostMap := configutils.GetNodeTypeMap(config)
+	for ip, types := range hostMap {
+		for i := 0; i < len(types); i++ {
+			var requestBody map[string]interface{}
```

Comment:
why do you need a map for the request?

Reply:
Removed the map, hence we don't need marshal and unmarshal